### PR TITLE
Issue 4259 - Fix double modification of rules - v2

### PR DESCRIPTION
**Pull request by jasonish**
Fix the case where only the last modification made to a rule takes effect. This
occurs when 2 modification lines change a rule, or a modification plus a drop
changes a rule.

Ticket: https://redmine.openinfosecfoundation.org/issues/4259

### CODE REVIEW

**Comment by victorjulien** [comment]
Looks like this is working as expected for my use case.

File: tests/integration_tests.py [hunk 1]
```diff
@@ -31,76 +35,188 @@ def delete(path):
...
+
+    def __init__(self):
+        IntegrationTest.__init__(self, self.configs)
+
+    def check(self):
+        # This rule should have been converted to drop.
+        rule1 = self.get_rule_by_sid(2103461)
+        assert(rule1.action == "drop")
+
+        # This one should have been converted back to
```
Comment (inashivb) [inline tests/integration_tests.py:197]: nit: missing `alert.` at the end

File: tests/integration_tests.py [hunk 2]
```diff
@@ -116,3 +118,105 @@ def delete(path):
...
+        IntegrationTest.__init__(self, self.configs)
+
+    def check(self):
+        rule1 = self.get_rule_by_sid(2024029)
+        assert(rule1.action == "drop")
+        assert(rule1.msg.startswith("TEST INFO"))
+
+
+MultipleModifyTest().run()
+DropAndModifyTest().run()
```
Comment (inashivb) [inline tests/integration_tests.py:222]: Neat code! :)

**Review by inashivb — APPROVED** [review]
Looks great to me! :)

File: tests/integration_tests.py (hunk 1 above)
Comment (jasonish) [reply tests/integration_tests.py:197]: Thanks, fixed in merge.

**Comment by jasonish** [comment]
Merged into master.

**Pull request closed without merging**